src/dom/Integer.ts

```diff
@@ -96,17 +96,25 @@ export class Integer extends Value(Number, IonTypes.INT, _fromJsConstructor) {
   ): boolean {
     let isSupportedType: boolean = false;
     let valueToCompare: any = null;
+
     if (options.onlyCompareIon) {
       // `compareOnlyIon` requires that the provided value be an ion.dom.Integer instance.
-      if (other instanceof Integer) {
```

Comment:
I think this `if` check was still needed to make sure that `isSupportedType = true` is right.